reference-lib/src/main/java/org/veriblock/integrations/rewards/PopRewardCalculator.java

```diff
@@ -139,6 +140,26 @@ public static int getBestPublicationHeight(List<AltPublication> publications) th
         return bestPublication;
     }
 
+    public static BigDecimal calculatePopDifficultyForBlock(List<AltChainBlock> blocksInterval) throws SQLException
+    {
+        BigDecimal difficulty = BigDecimal.ZERO;
+
+        Collections.sort(blocksInterval);  // make the ascending order for the blocks in the collection, it needs for the correct calculation of the pop score
+
+        for(int i = 0; i < blocksInterval.size(); ++i)
+        {
+            BigDecimal score = calculatePopScoreFromEndorsements(blocksInterval.get(i), blocksInterval.subList(i , blocksInterval.size()));
```

Comment:
Each of the blocks should have own endorsements list. It may happen that passed list of the endorsements does not contain the endorsement for previous blocks.

Reply:
And if the block has zero endorsments, the score will be zero. As I understand it is a possible situation. Or what do you mean ?